nodejs/private/ts_validate_options.bzl

```diff
@@ -56,32 +62,16 @@ def _validate_options_impl(ctx):
         ValidOptionsInfo(marker = marker),
     ]
 
-# These attrs are shared between the validate and the ts_project rules
-SHARED_ATTRS = {
-    "allow_js": attr.bool(),
-    "composite": attr.bool(),
-    "declaration": attr.bool(),
-    "declaration_map": attr.bool(),
-    "emit_declaration_only": attr.bool(),
-    "extends": attr.label(allow_files = [".json"]),
-    "incremental": attr.bool(),
-    "preserve_jsx": attr.bool(),
-    "resolve_json_module": attr.bool(),
-    "source_map": attr.bool(),
-}
-
-validate_options = rule(
-    implementation = _validate_options_impl,
-    attrs = dict(SHARED_ATTRS, **{
-        "has_local_deps": attr.bool(doc = "Whether any of the deps are in the local workspace (not starting with '@')"),
-        "target": attr.string(),
-        "ts_build_info_file": attr.string(),
-        "tsconfig": attr.label(mandatory = True, allow_single_file = [".json"]),
-        "validator": attr.label(default = Label("//packages/typescript/bin:ts_project_options_validator"), executable = True, cfg = "exec"),
-    }),
-)
+_ATTRS = dict(COMPILER_OPTION_ATTRS, **{
+    "has_local_deps": attr.bool(doc = "Whether any of the deps are in the local workspace (not starting with '@')"),
```

Comment:
technically `@//path/to/local:file` is in the local workspace and starts with `@`

Reply:
that's true, `_is_external_label` is what we use to set this attribute, which is just
`len(_to_label(param).workspace_root) > 0`